mobilecoinfoundation/mobilecoin
# Fix CI cache management on master

### Motivation

This PR attempts to address CI build issues that are currently being experienced on master (see https://app.circleci.com/pipelines/github/mobilecoinfoundation/mobilecoin/3153/workflows/201682aa-9860-4a1f-a783-d2ab0f99a936).

### In this PR
Changelog:
* Rename `record-cargo-cache-stats` to `print-cargo-cache-stats`
* Remove unused jobs
* Remove unnecessary parallelism key when setting to 1
* Remove unused `test-executor`
* Add `rustup show` to CI Version Info printout
* Update Cargo deps on master right after installing them
* Ensure cargo install in CI uses Rust stable
* Move cargo-cache saving to end of job
* Add documentation comment to cache save jobs
* Print cargo cache stats for all jobs
* Allow cargo cache local command to fail
